### PR TITLE
Texture2D::TexParams fields use backend enum types

### PR DESCRIPTION
**Pull request by PatriceJiang**
ref https://github.com/cocos2d/cocos2d-x/issues/19334

### CODE REVIEW

**Comment by minggo** [comment]
But `Texture2D::setSamplerDescriptor()` is not deleted.

**Comment by PatriceJiang** [comment]
alias `backend::SamplerDescriptor` as `Texture2D::TexParams`

File: cocos/renderer/backend/Types.h [hunk 1]
```diff
@@ -188,21 +188,32 @@ enum class ColorWriteMask: uint32_t
 
 struct SamplerDescriptor
 {
-    bool mipmapEnabled = false;
+    bool mipmapEnabled = true;
```
Comment (minggo) [inline cocos/renderer/backend/Types.h]: Is the default value need to be true?

File: cocos/renderer/CCTexture2D.cpp [hunk 2]
```diff
@@ -707,21 +707,20 @@ void Texture2D::setAliasTexParameters()
...
-
-    setSamplerDescriptor(descriptor);
+    _texture->updateSamplerDescriptor(descriptor);
 }
 
 void Texture2D::setAntiAliasTexParameters()
 {
 
-    backend::SamplerDescriptor descriptor(false,
+    backend::SamplerDescriptor descriptor(true,
```
Comment (PatriceJiang) [inline cocos/renderer/CCTexture2D.cpp]:
```suggestion
    backend::SamplerDescriptor descriptor(false,
```

File: cocos/renderer/backend/Types.h (hunk 1 above)
Comment (PatriceJiang) [inline cocos/renderer/backend/Types.h]:
```suggestion
    bool mipmapEnabled = false;
```

**Review by minggo — APPROVED** [review]

**Pull request merged**